src/main/java/com/google/sps/servlets/GetBlurAreasServlet.java

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.FaceAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.Vertex;
+import com.google.gson.Gson;
+import com.google.protobuf.ByteString;
+import java.awt.Point;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-blur-areas")
+public class GetBlurAreasServlet extends HttpServlet {
+
+  // Image types that are supported by our application.
+  private static final ArrayList<String> supportedTypes =
+      new ArrayList<>(Arrays.asList("image/jpeg", "image/png"));
+
+  /**
+   * This method handles the POST requests to "/get-blur-areas". Receives a BlobKey parameter which
+   * represents the Blobstore location of an image. Responds with a JSON ArrayList of rectangles. A
+   * rectangle is represented by a List of points. A point contains two properties, its coordinates:
```

Comment:
I am returning them in the order given by the Cloud Vision API and it doesn't guarantee any order. I can order them if needed, but currently this is being done on the front-end. What do you think?

Reply:
Then a note about 'no guarantees about the order of points' in the comment would be helpful. Since another part of the service takes the list as an input it should know about the order (I assume). Therefore you could either sort the points, and make the comment about the order, or say that the order is unknown and let the other part deal with it. In either case the documentation for your method should make it clear for others who are going to use it.

Reply:
Ah, yea, right, I think saying that the order is not known is the best way to deal with this. Thank you!